ext/dapr-ext-workflow/dapr/ext/workflow/aio/dapr_workflow_client.py

```diff
@@ -0,0 +1,280 @@
+from dapr.clients.http.client import DAPR_API_TOKEN_HEADER
+from dapr.conf import settings
+from dapr.conf.helpers import GrpcEndpoint
+
+T = TypeVar('T')
+TInput = TypeVar('TInput')
+TOutput = TypeVar('TOutput')
+
+
+class DaprWorkflowClient:
+    """Async client for managing Dapr Workflow instances.
+
+    This uses a gRPC async connection to send commands directly to the workflow engine,
+    bypassing the Dapr API layer. Intended to be used by workflow applications.
+    """
+
+    def __init__(
+        self,
+        host: Optional[str] = None,
+        port: Optional[str] = None,
+        logger_options: Optional[LoggerOptions] = None,
+    ):
+        address = getAddress(host, port)
+
+        try:
+            uri = GrpcEndpoint(address)
+        except ValueError as error:
+            raise DaprInternalError(f'{error}') from error
+
+        self._logger = Logger('DaprWorkflowClient', logger_options)
+
+        metadata = tuple()
+        if settings.DAPR_API_TOKEN:
+            metadata = ((DAPR_API_TOKEN_HEADER, settings.DAPR_API_TOKEN),)
+        options = self._logger.get_options()
+        self.__obj = aioclient.AsyncTaskHubGrpcClient(
+            host_address=uri.endpoint,
+            metadata=metadata,
+            secure_channel=uri.tls,
+            log_handler=options.log_handler,
+            log_formatter=options.log_formatter,
+        )
+
+    async def schedule_new_workflow(
+        self,
+        workflow: Workflow,
+        *,
+        input: Optional[TInput] = None,
+        instance_id: Optional[str] = None,
+        start_at: Optional[datetime] = None,
+        reuse_id_policy: Optional[pb.OrchestrationIdReusePolicy] = None,
+    ) -> str:
+        """Schedules a new workflow instance for execution.
+
+        Args:
+            workflow: The workflow to schedule.
+            input: The optional input to pass to the scheduled workflow instance. This must be a
+            serializable value.
+            instance_id: The unique ID of the workflow instance to schedule. If not specified, a
+            new GUID value is used.
+            start_at: The time when the workflow instance should start executing.
+            If not specified or if a date-time in the past is specified, the workflow instance will
+            be scheduled immediately.
+            reuse_id_policy: Optional policy to reuse the workflow id when there is a conflict with
+            an existing workflow instance.
+
+        Returns:
+            The ID of the scheduled workflow instance.
+        """
+        if hasattr(workflow, '_dapr_alternate_name'):
+            return await self.__obj.schedule_new_orchestration(
+                workflow.__dict__['_dapr_alternate_name'],
+                input=input,
+                instance_id=instance_id,
+                start_at=start_at,
+                reuse_id_policy=reuse_id_policy,
+            )
+        return await self.__obj.schedule_new_orchestration(
+            workflow.__name__,
```

Comment:
Minor - consider storing the `workflow.__dict__['_dapr_alternate_name']` / `workflow.__name__` in a variable and avoiding code duplication around calling `self.__obj.schedule_new_orchestration`